imas/test/test_helpers.py

```diff
@@ -269,10 +313,53 @@ def fill_consistent(
             elif any(len(coordinate.references) > 1 for coordinate in coordinates):
                 exclusive_coordinates.append(child)
             else:
-                try:
-                    maybe_set_random_value(child, leave_empty, skip_complex)
-                except (RuntimeError, ValueError):
-                    pass
+                same_as_skip = False
+                for dim, same_as in enumerate(child.metadata.coordinates_same_as):
+                    if same_as.references:
+                        try:
+                            ref_elem = same_as.references[0].goto(child)
+                            if len(ref_elem.shape) <= dim or ref_elem.shape[dim] == 0:
+                                same_as_skip = True
+                                break
+                        except (ValueError, AttributeError, IndexError, RuntimeError):
+                            same_as_skip = True
+                            break
+
+                error_skip = False
+                if child.metadata.name.endswith("_error_upper"):
+                    name = child.metadata.name[: -len("_error_upper")]
+                    try:
+                        data = child._parent[name]
+                        if not data.has_value:
+                            maybe_set_random_value(data, 0.0, skip_complex)
+                        if (
+                            not data.has_value
+                            or len(data.shape) == 0
+                            or any(s == 0 for s in data.shape)
+                        ):
+                            error_skip = True
+                    except (KeyError, AttributeError, RuntimeError, ValueError):
+                        error_skip = True
+                elif child.metadata.name.endswith("_error_lower"):
+                    name = child.metadata.name[: -len("_error_lower")] + "_error_upper"
+                    try:
+                        data = child._parent[name]
+                        if not data.has_value:
+                            maybe_set_random_value(data, 0.0, skip_complex)
+                        if (
+                            not data.has_value
+                            or len(data.shape) == 0
+                            or any(s == 0 for s in data.shape)
+                        ):
+                            error_skip = True
+                    except (KeyError, AttributeError, RuntimeError, ValueError):
+                        error_skip = True
```

Comment:
This logic looks almost identical to the changes on lines +120 to +160. Do we need both?